### PR TITLE
[FLINK-15231][table-planner-blink] Do not support TIMESTAMP/DECIMAL t…

### PR DESCRIPTION
**Pull request by docete**
…ypes for HeapVector

## What is the purpose of the change

For TIMESTAMP WITHOUT TIME ZONE/TIMESTAMP WITH LOCAL TIME ZONE/DECIMAL types, AbstractHeapVector.createHeapColumn generates wrong HeapVectors. This PR removes TIMESTAMP/TIMESTAMP WITH LOCAL TIME ZONE/DECIMAL support since the HeapVectors is used for testing.

## Brief change log

- dc4897 do not support TIMESTAMP/DECIMAL types for HeapVector

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (**yes** / no)
  - If yes, how is the feature documented? (not applicable / docs / **JavaDocs** / not documented)

### CODE REVIEW

File: flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/vector/heap/AbstractHeapVector.java [hunk 1]
```diff
@@ -109,18 +107,7 @@ public static AbstractHeapVector createHeapColumn(LogicalType fieldType, int max
 			case TIME_WITHOUT_TIME_ZONE:
 				return new HeapIntVector(maxRows);
 			case BIGINT:
-			case TIMESTAMP_WITHOUT_TIME_ZONE:
```
Comment (KurtYoung) [inline flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/vector/heap/AbstractHeapVector.java]: looks like both `allocateHeapVectors` and `createHeapColumn` are useless?

Comment (docete) [reply flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/vector/heap/AbstractHeapVector.java]: Uses should use `allocateHeapVectors` or `createHeapColumn` to create HeapVectors. See the java doc of constructor of `HeapIntVector`. Uses should not call `new HeapIntVector(len)` except for testing purposes.

Comment (KurtYoung) [reply flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/vector/heap/AbstractHeapVector.java]: What kind of user you are talking about?

Comment (docete) [reply flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/vector/heap/AbstractHeapVector.java]: May be some connectors? The classes in `org.apache.flink.table.dataformat.vector.heap` is not used except testing now, but may be used to develop some connectors? Or if the aim of these classes is only for testing, I could move then to `test` and remove the two functions.
What do you think?

**Comment by docete** [comment]
remove useless methods

**Pull request merged**